Vector_graphics_on_surfaces/include/CGAL/Vector_graphics_on_surfaces/locally_shortest_path.h

```diff
@@ -0,0 +1,4814 @@
+      : destination_vd(destination_vd)
+    { }
+
+    void initialize_vertex(const vertex_descriptor& /*s*/, const Graph& /*mesh*/) const { }
+    void examine_vertex(const vertex_descriptor& /*s*/, const Graph& /*mesh*/) const { }
+    void examine_edge(const edge_descriptor& /*e*/, const Graph& /*mesh*/) const { }
+    void edge_relaxed(const edge_descriptor& /*e*/, const Graph& /*mesh*/) const { }
+    void discover_vertex(const vertex_descriptor& /*s*/, const Graph& /*mesh*/) const { }
+    void edge_not_relaxed(const edge_descriptor& /*e*/, const Graph& /*mesh*/) const { }
+    void finish_vertex(const vertex_descriptor &vd, const Graph& /* mesh*/) const
+    {
+      if(vd == destination_vd)
+        throw Dijkstra_end_exception();
+    }
+  };
+#endif
+
+
+} // namespace internal
+
+#ifndef CGAL_BSURF_USE_DIJKSTRA_SP
+/*!
+ * \ingroup VGSMiscellaneous
+ * Geodesic solver class used to store precomputed information of a given mesh for
+ * approximate geodesic compution. Those information are computed by the function
+ * `init_geodesic_dual_solver()`.
+ * \tparam FT floating point number type (float or double)
+ */
+template <class FT>
+struct Dual_geodesic_solver
+{
+  struct Edge {
+    int node = -1;
+    FT len = DBL_MAX;
+  };
+  std::vector<std::array<Edge, 3>> graph = {};
+};
+
+/*!
+ * \ingroup VGSMiscellaneous
+ * fills `solver` for a given mesh `tmesh`. It is the user responsability to
+ * call again this function if `tmesh` or the points of its vertices are modified.
+ * If `solver` was used in a previous call to this function, information will be ovewritten.
+ * \tparam TriangleMesh a model of `FaceListGraph` and `EdgeListGraph`
+ * \tparam FT floating point number type (float or double)
+ * \param solver the container for the precomputed information
+ * \param tmesh triangle mesh to be considered for the precomputations
+ * \todo add named parameters
+ * \todo make sure solver.graph is cleared before filling it
+ */
+template <class FT, class TriangleMesh>
+void init_geodesic_dual_solver(Dual_geodesic_solver<FT>& solver, const TriangleMesh& tmesh)
+{
+  //TODO replace with named parameter
+  using VPM = typename boost::property_map<TriangleMesh, CGAL::vertex_point_t>::const_type;
+  using K =  typename Kernel_traits<typename boost::property_traits<VPM>::value_type>::type;
+  VPM vpm = get(CGAL::vertex_point, tmesh);
+  typedef typename GetInitializedFaceIndexMap<TriangleMesh, parameters::Default_named_parameters>::const_type FIM;
+  typedef typename GetInitializedVertexIndexMap<TriangleMesh, parameters::Default_named_parameters>::const_type VIM;
+  const FIM fim = get_initialized_face_index_map(tmesh, parameters::default_values());
+
+  using Impl2 = typename internal::Geodesic_circle_impl<K, TriangleMesh, VPM, VIM, FIM>;
+  solver=Impl2::make_dual_geodesic_solver(vpm, fim, tmesh);
+}
+#endif
+
+/*!
+ * \ingroup VGSFunctions
+ * computes an approximated geodesic shortest path between two locations on a
```

Comment:
The definition of _geodesic_ on the[ wikipedia](https://en.wikipedia.org/wiki/Geodesic) says that it is the shortewst path.

Reply:
it is to indicate that it's not on the graph but on the surface. It is also mentioned like that in "Triangulated Surface Mesh Shortest Paths"